embedded-graphics/src/primitives/mod.rs

```diff
@@ -243,23 +243,36 @@ macro_rules! egline {
 ///     .build();
 ///
 /// let rectangle_2: Styled<Rectangle, PrimitiveStyle<Rgb565>> =
-///     Rectangle::new(Point::new(10, 20), Point::new(30, 40)).into_styled(style);
+///     Rectangle::new(Point::new(10, 20), Size::new(20, 20)).into_styled(style);
 ///
 /// assert_eq!(rectangle_1, rectangle_2);
 /// ```
 #[macro_export]
 macro_rules! egrectangle {
-    (top_left = $top_left:expr, bottom_right = $bottom_right:expr $(,)?) => {{
+    (corners = [$corner_1:expr, $corner_2:expr] $(,)?) => {{
+        $crate::egrectangle!(
+            corners = [$corner_1, $corner_2],
+            style = $crate::style::PrimitiveStyle::default()
+        )
+    }};
+    (corners = [$corner_1:expr, $corner_2:expr], style = $style:expr $(,)?) => {{
+        $crate::primitives::Rectangle::with_corners(
+            $crate::geometry::Point::from($corner_1),
+            $crate::geometry::Point::from($corner_2),
+        )
+        .into_styled($style)
+    }};
```

Comment:
Would `box` or `bounding_box` be a better name perhaps? The `fn` version is called `from_bounding_box` so some consistency with that might be good. Naming this param is pretty hard.

Reply:
I want to reserve the `bounding_box` name for something else, so I went with corners for this.